pkg/controller/garbagecollector/patch.go

```diff
@@ -39,21 +39,27 @@ func deleteOwnerRefStrategicMergePatch(dependentUID types.UID, ownerUIDs ...type
 	return []byte(patch)
 }
 
-// TODO: remove this function when we can use strategic merge patch
 func (gc *GarbageCollector) getCachedMetadata(apiVersion, kind, namespace, name string) (metav1.Object, error) {
 	fqKind := schema.FromAPIVersionAndKind(apiVersion, kind)
 	mapping, err := gc.restMapper.RESTMapping(fqKind.GroupKind(), fqKind.Version)
 	if err != nil {
 		return nil, newRESTMappingError(kind, apiVersion)
 	}
-	gvr := schema.GroupVersionResource{Group: fqKind.Group, Version: fqKind.Version, Resource: mapping.Resource}
 	gc.dependencyGraphBuilder.monitorLock.RLock()
 	defer gc.dependencyGraphBuilder.monitorLock.RUnlock()
-	m, ok := gc.dependencyGraphBuilder.monitors[gvr]
+	m, ok := gc.dependencyGraphBuilder.monitors[mapping.Resource]
 	if !ok || m == nil {
-		// TODO: consider to make a call to apiserver in case the local cache is
-		// unavailable.
-		return nil, fmt.Errorf("doesn't have a local cache for %s", gvr)
+		// If local cache doesn't exist for mapping.Resource, send a GET request to API server
+		apiResource, _, err := gc.apiResource(apiVersion, kind)
```

Comment:
why not use mapping.Resource again? looks like it does the same thing as the beginning of this function https://github.com/roycaihw/kubernetes/blob/ebb54562d1789f0e198cad35ad34a0ae1175deb4/pkg/controller/garbagecollector/operations.go#L43-L50

Reply:
Nice catch! I guess I was copypasting something and didn't realize the redundant.